fix capitalization rules (#15544)
dotnet/docs: styleguide/template.md

## Headings

> Use sentence-style capitalization. Always capitalize the first word of a heading, including the word following a colon in a title or heading (for example, "Tutorial: Predict prices using regression with ML.NET"). Don't add a colon after How to (for example, "How to sort an array").

Reviewer: ```suggestion
Use sentence-style capitalization. Capitalize the first letter of the first word of a heading, proper nouns, and the first letter following a colon (for example, "Tutorial: Predict prices using regression with ML.NET").

Don't add a colon after "How to" (for example, "How to sort an array" and not "How to: Sort an array").
```
